packages/graphql/src/services/sessions.ts

```diff
@@ -362,17 +360,82 @@ export async function deactivateSessionBlock(
   )
 
   const redisMulti = ctx.redisExec.multi()
+  redisMulti.hgetall(`s:${sessionId}:lb`)
+  redisMulti.hgetall(`s:${sessionId}:b:${sessionBlockId}:lb`)
   activeInstanceIds.forEach((instanceId) => {
-    redisMulti.hgetall(`s${sessionId}:i:${instanceId}:results`)
+    redisMulti.hgetall(`s:${sessionId}:i:${instanceId}:responseHashes`)
+    redisMulti.hgetall(`s:${sessionId}:i:${instanceId}:responses`)
+    redisMulti.hgetall(`s:${sessionId}:i:${instanceId}:results`)
   })
-  const allInstanceResults = await redisMulti.exec()
+  const cachedResults = await redisMulti.exec()
 
-  console.warn(allInstanceResults)
+  if (!cachedResults) return session
 
-  return session
+  const mappedResults: any[] = cachedResults.map(([_, result]) => result)
+
+  const sessionLeaderboard: Record<string, string> = mappedResults[0]
+  const blockLeaderboard: Record<string, string> = mappedResults[1]
+
+  const instanceResults: Record<
```

Comment:
this is critical and a bit complex. the redismulti above returns everything that has been added to the "transaction" in one large array, so e.g., [session leaderboard, session block leaderboard, instance1 response hashes, instance 1 responses, instance 1 results, ...].

all of these items have a different structure, hence cutting off the first two separately, and using a modulo-based approach to select the first, second, and third items of each instance